src/robot.coffee

```diff
@@ -247,6 +264,28 @@ class Robot
   # Returns nothing.
   # Returns before executing callback
   receive: (message, cb) ->
+    # When everything is finished (down the middleware stack and back up),
+    # pass control back to the robot
+    receiveMiddlewareDone = () ->
+      if cb?
+        process.nextTick -> cb true
+
+    @middleware.receive.execute(
+      {response: new Response(this, message)}
```

Comment:
I don't think we should be creating a `Response` at this stage. The current model is a unique `Response` object for each matched `Listener`. I now see the problem there though: no `res.reply`, etc. Hmm....